sdks/go/pkg/beam/transforms/stats/count.go

```diff
@@ -18,18 +18,36 @@ package stats
 
 import (
 	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/typex"
 )
 
-// Count counts the number of elements in a collection. It expects a
-// PCollection<T> as input and returns a PCollection<KV<T,int>>. T's encoding
-// must be a well-defined injection.
+// Count counts the number of appearances of each element in a collection. It
+// expects a PCollection<T> as input and returns a PCollection<KV<T,int>>. T's
+// encoding must be a well-defined injection.
 func Count(s beam.Scope, col beam.PCollection) beam.PCollection {
 	s = s.Scope("stats.Count")
 
-	pre := beam.ParDo(s, mapFn, col)
+	pre := beam.ParDo(s, keyedMapFn, col)
 	return SumPerKey(s, pre)
 }
 
-func mapFn(elm beam.T) (beam.T, int) {
+func keyedMapFn(elm beam.T) (beam.T, int) {
 	return elm, 1
 }
+
+// CountElms counts the number of elements in a collection. It expects a
+// PCollection<T> as input and returns a PCollection<int> of one element
+// containing the count. T's encoding must be a well-defined injection.
+func CountElms(s beam.Scope, col beam.PCollection) beam.PCollection {
+	s = s.Scope("stats.CountElms")
+
+	if typex.IsKV(col.Type()) {
+		col = beam.DropKey(s, col)
+	}
+	pre := beam.ParDo(s, mapFn, col)
+	return Sum(s, pre)
+}
+
+func mapFn(_ beam.T) int {
```

Comment:
Consider countFn?

Reply:
Sounds good to me. Changed this to countFn and the other to keyedCountFn.